### PR TITLE
feat(j-s): SQS Setup

### PR DESCRIPTION
**Pull request by gudjong**
# SQS Setup

https://app.asana.com/0/1199153462262248/1201777996978577/f

## What

- Creates a new Nest application for processing messages from a dedicated judicial system SQS queue.
- Updates Helm charts to deploy the new Nest application.

## Why

This will facilitate the decoupling of backend processing, such as sending notifications and uploading files.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/infra/judicial-system-backend.ts [hunk 1]
```diff
@@ -54,6 +54,10 @@ export const serviceSetup = (): ServiceBuilder<'judicial-system-backend'> =>
         staging: 'https://judicial-system.staging01.devland.is/verjandi/',
         prod: 'https://rettarvorslugatt.island.is/verjandi/',
       },
+      SQS_QUEUE_NAME: 'sqs-judicial-system',
+      SQS_DEAD_LETTER_QUEUE_NAME: 'sqs-judicial-system-dlq',
+      SQS_REGION: 'eu-west-1',
+      SQS_ENABLED: { dev: 'true', staging: 'false', prod: 'false' },
```
Comment (pshomov) [inline apps/judicial-system/backend/infra/judicial-system-backend.ts:60]: We already have server-side feature toggles that seem should fit your usage - https://docs.devland.is/technical-overview/devops/service-setup#server-side-feature-flags

File: apps/judicial-system/message-handler/infra/judicial-system-message-handler.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    .env({
+      SQS_QUEUE_NAME: 'sqs-judicial-system',
+      SQS_DEAD_LETTER_QUEUE_NAME: 'sqs-judicial-system-dlq',
+      SQS_REGION: 'eu-west-1',
+      BACKEND_URL: ref((h) => `http://${h.svc(services.backend)}`),
+    })
+    .secrets({
+      BACKEND_ACCESS_TOKEN: '/k8s/judicial-system/BACKEND_ACCESS_TOKEN',
+    })
+    .replicaCount({ min: 1, max: 1, default: 1 })
```
Comment (pshomov) [inline apps/judicial-system/message-handler/infra/judicial-system-message-handler.ts:18]: Why only one replica?

File: apps/judicial-system/backend/infra/judicial-system-backend.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/backend/infra/judicial-system-backend.ts]: Switched to using server-side feature toggles. Not sure its correctly done thought.

File: apps/judicial-system/message-handler/infra/judicial-system-message-handler.ts (hunk 2 above)
Comment (gudjong) [reply apps/judicial-system/message-handler/infra/judicial-system-message-handler.ts:18]: No real reason. I will remove this.

File: apps/judicial-system/backend/src/app/modules/case/case.controller.ts [hunk 3]
```diff
@@ -222,6 +235,15 @@ export class CaseController {
       state !== CaseState.DELETED,
     )
 
+    if (
+      updatedCase &&
+      completedCaseStates.includes(updatedCase.state) &&
+      ServerSideFeatureClient.isOn('judicial-system-sqs')
```
Comment (saevarma) [inline apps/judicial-system/backend/src/app/modules/case/case.controller.ts:241]: You should be using the `FeatureFlagService` from the `nest/feature-flags` library, it also takes in the user object to check if a specific user has access to the feature.

File: infra/src/uber-charts/judicial-system.ts [hunk 4]
```diff
@@ -11,9 +12,10 @@ const jsApi = jsApiSetup({ backend: jsBack })
 const jsWeb = jsWebSetup({ api: jsApi })
 const jsXrdApi = jsXrdApiSetup({ backend: jsBack })
 const jsScheduler = jsSchedulerSetup({ backend: jsBack })
+const jsMessageHndler = jsMessageHandlerSetup({ backend: jsBack })
```
Comment (saevarma) [inline infra/src/uber-charts/judicial-system.ts:15]:
```suggestion
const jsMessageHandler = jsMessageHandlerSetup({ backend: jsBack })
```

**Review by saevarma — CHANGES_REQUESTED** [review]

**Review by Arnarkari93 — APPROVED** [review]

File: apps/judicial-system/backend/src/app/modules/case/case.controller.ts (hunk 3 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/case.controller.ts:241]: I am using the `@island.is/feature-flags`. Why is that not good enough?

**Review by pshomov — APPROVED** [review]
LGTM

**Review by saevarma — APPROVED** [review]

**Pull request merged**